modules/environment/aws/code-services/main.tf

```diff
@@ -202,3 +202,52 @@ resource "aws_sqs_queue_policy" "code_services_queue_policy" {
 }
 POLICY
 }
+
+resource "aws_iam_role" "sqs_role" {
+  count  = var.enable_aws_code_services ? 1 : 0
+  name = "${var.cluster}_sqs_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${var.account_id}:oidc-provider/${replace(var.openid_connect_provider_url, "https://", "")}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${replace(var.openid_connect_provider_url, "https://", "")}:aud": "sts.amazonaws.com"
+        }
+      }
+    }
+  ]
+}
+EOF
+
+  permissions_boundary = "arn:aws:iam::${var.account_id}:policy/Developer"
+}
+
+resource "aws_iam_role_policy" "sqs_role_policy" {
+  count  = var.enable_aws_code_services ? 1 : 0
+  name   = "${var.cluster}_sqs_role_policy"
+  role   = aws_iam_role.sqs_role[0].name
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+        "Action": [
+            "sqs:*",
+            "sts:*"
```

Comment:
I think it's alright to dev with these permissions but I really don't think this should be merged until this is trimmed down.